src/api/middleware/userRoleMiddleware.ts

```diff
@@ -0,0 +1,42 @@
+import { Handler } from 'express';
+
+import { UserRoleId } from '../entities/UserRole';
+import { UserToParticipantRole } from '../entities/UserToParticipantRole';
+import { ParticipantRequest } from '../services/participantsService';
+import { findUserByEmail } from '../services/usersService';
+
+export const isUid2Support = async (userEmail: string) => {
+  const user = await findUserByEmail(userEmail);
+  const userWithUid2SupportRole = await UserToParticipantRole.query()
+    .where('userId', user!.id)
+    .andWhere('userRoleId', UserRoleId.UID2Support)
+    .first();
+  return !!userWithUid2SupportRole;
+};
+
+export const isUid2SupportCheck: Handler = async (req: ParticipantRequest, res, next) => {
+  if (!(await isUid2Support(req.auth?.payload?.email as string))) {
+    return res.status(403).json({
+      message: 'Unauthorized. You do not have the necessary permissions.',
+      errorHash: req.headers.traceId,
+    });
+  }
+  next();
+};
+
+export const isAdminOrUid2SupportCheck: Handler = async (req: ParticipantRequest, res, next) => {
```

Comment:
It feels a little redundant to have `isUid2Support` and `isAdminOrUid2Support` - my opinion would be to have this function be `isAdminCheck` and then when you need both, you'd call `isUid2SupportCheck` || `isAdminCheck`.

Reply:
As I mentioned in stand-up, I can't find a good way to enforce this conditional logic in handlers. Using `||` doesn't really work because if the first check succeeds then we just call `next()` which seems to skip the next check. Alternatively, placing it as an additional handler argument makes it an "and" condition which isn't what we want. 

I'll leave as-is for now.